XpringKit/Utils.swift

```diff
@@ -1,66 +1,132 @@
+    }
 
-  /// Decode a classic address from a given x-address.
-  ///
-  /// - SeeAlso: https://xrpaddress.info/
-  ///
-  /// - Parameter xAddress: The xAddress to decode.
-  /// - Returns: a tuple containing the decoded address,  tag and bool indicating if the address was on a test network.
-  public static func decode(xAddress: Address) -> (classicAddress: String, tag: UInt32?, isTest: Bool)? {
-    return javaScriptUtils.decode(xAddress: xAddress)
-  }
+    /// Decode a classic address from a given x-address.
+    ///
+    /// - SeeAlso: https://xrpaddress.info/
+    ///
+    /// - Parameter xAddress: The xAddress to decode.
+    /// - Returns: a tuple containing the decoded address,  tag and bool indicating if the address was on a test network.
+    public static func decode(xAddress: Address) -> (classicAddress: String, tag: UInt32?, isTest: Bool)? {
+        return javaScriptUtils.decode(xAddress: xAddress)
+    }
 
-  /// Convert the given transaction blob to a transaction hash.
-  ///
-  /// - Parameter transactionBlobHex: A hexadecimal encoded transaction blob.
-  /// - Returns: A hex encoded hash if the input was valid, otherwise undefined.
-  public static func toTransactionHash(transactionBlobHex: Hex) -> Hex? {
-    return javaScriptUtils.toTransactionHash(transactionBlobHex: transactionBlobHex)
-  }
+    /// Convert the given transaction blob to a transaction hash.
+    ///
+    /// - Parameter transactionBlobHex: A hexadecimal encoded transaction blob.
+    /// - Returns: A hex encoded hash if the input was valid, otherwise undefined.
+    public static func toTransactionHash(transactionBlobHex: Hex) -> Hex? {
+        return javaScriptUtils.toTransactionHash(transactionBlobHex: transactionBlobHex)
+    }
+
+    /// Convert the given string to a bytes array UInt8.
+    ///
+    /// - Parameter hex: A string.
+    /// - Returns: A byte array in [UInt8].
+    public static func toByteArray(hex: String ) -> [UInt8] {
+        // remove "-" from Hexadecimal
+        let hexString = hex.removeWord( "-" )
+        let size = hexString.count / 2
+        var result: [UInt8] = [UInt8]( repeating: 0, count: size ) // array with length = size
+        for i in stride( from: 0, to: hexString.count, by: 2 ) {
+            let subHexStr = hexString.subString( i, length: 2 )
+            guard let _uint8 = UInt8( subHexStr, radix: 16 ) else {
+                return []
+            }
+            result[ i / 2 ] = _uint8
+        }
+        return result
+    }
+}
+extension String {
+
+    func subString( _ from: Int, length: Int ) -> String {
+        let size = self.count
+        let to = length + from
+        if from < 0 || to > size {
+            return ""
+        }
+        var result = ""
+        for ( idx, char ) in self.enumerated() {
+            if idx >= from && idx < to {
+                result.append( char )
+            }
+        }
+        return result
+    }
+
+    func removeWord( _ word: String ) -> String {
```

Comment:
It's a bit opaque what this method is doing. I think that you could just use [dropFirst](https://developer.apple.com/documentation/swift/string/2893343-dropfirst).